src/Modal/ModalContents.js

```diff
@@ -222,34 +223,52 @@ export default class ModalContents extends Util.mixin(BindMixin) {
     }
 
     return (
-      <div
-        className={props.containerClass}
-        style={containerStyle}>
-        <div ref="modal" className={props.modalClass}>
-          {this.getCloseButton()}
-          {this.getHeader()}
-          <div className={props.bodyClass} style={modalStyle}>
-            <div ref="innerContainer" className={props.innerBodyClass}>
-              {this.getModalContent(useScrollbar, contentHeight)}
-            </div>
+      <div ref="modal" className={props.modalClass}>
+        {this.getCloseButton()}
+        {this.getHeader()}
+        <div className={props.bodyClass} style={modalStyle}>
+          <div ref="innerContainer" className={props.innerBodyClass}>
+            {this.getModalContent(useScrollbar, contentHeight)}
           </div>
-          {this.getFooter()}
-        </div>
-        <div className={props.backdropClass}
-          onClick={this.handleBackdropClick}>
         </div>
+        {this.getFooter()}
       </div>
     );
   }
 
+  getBackdrop() {
+    let props = this.props;
+    if (!props.open) {
+      return null;
+    }
+
+    return (
+      <div className={props.backdropClass} onClick={this.handleBackdropClick} />
+    );
+  }
+
   render() {
+    let props = this.props;
+    let containerClass = {};
+    if (props.open) {
+      containerClass[props.containerClass] = true;
```

Comment:
It has to change, because we can't apply the classes when it is not open. Otherwise it will cover over the whole page all the time